src/proto/colors.rs

```diff
@@ -0,0 +1,45 @@
+//! An extension trait that provides the ability to strip IRC colors from a string
+use regex::Regex;
+
+/// An extension trait giving strings a function to strip IRC colors
+pub trait FormattedString {
+
+    /// Returns true if the string contains color, bold, underline or italics
+    fn is_formatted(&self) -> bool;
+
+    /// Returns the string with all color, bold, underline and italics stripped
+    fn strip_colors(&self) -> String;
+
+    const STRIP_COLORS: Regex = Regex::new(r"\x1F|\x02|\x12|\x0F|\x16|\x03(?:\d{1,2}(?:,\d{1,2})?)?").unwrap();
+
+    const BOLD: char = '\x02';
```

Comment:
I think of these constants as being implementation details rather than part of the interface. Do you think think there's a reason to expose them publcily? If not, I would move them outside of the trait definition.